### PR TITLE
feat: new SPI to replace direct API client usage in modeler and meditor

### PR DESCRIPTION
**Pull request by xMort**
JIRA: LX-146

---

> [!IMPORTANT]
> Please, **don't forget to run `rush change`** for the commits that introduce **new features** 🙏

---

Refer to [documentation](https://github.com/gooddata/gooddata-ui-sdk/blob/master/dev_docs/continuous_integration.md) to see how to run checks and tests in the pull request. This is the list of the most used commands:

```
extended test - backstop
```

```
extended test - tiger-cypress - integrated
extended test - tiger-cypress - isolated
extended test - tiger-cypress - record
```

### CODE REVIEW

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/18745/ : SUCCESS in 3s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/20433/ : SUCCESS in 10m 02s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/18754/ : SUCCESS in 3s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/20447/ : SUCCESS in 9m 59s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/18756/ : SUCCESS in 3s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/20449/ : SUCCESS in 10m 52s

File: libs/sdk-backend-tiger/src/backend/workspace/dataFilters/index.ts [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+import { TigerAuthenticatedCallGuard } from "../../../types/index.js";
+import { JsonApiWorkspaceDataFilterSettingOutWithLinks } from "@gooddata/api-client-tiger";
+
+export class TigerDataFiltersService implements IDataFiltersService {
+    constructor(private readonly authCall: TigerAuthenticatedCallGuard, public readonly workspace: string) {}
+
+    public async getWorkspaceDataFilters(): Promise<IWorkspaceDataFilter[]> {
+        return this.authCall(async (client) => {
+            const [entitiesResult, settingsResult] = await Promise.all([
+                client.entities.getAllEntitiesWorkspaceDataFilters({
```
[the remark this answers is not in the document]
Comment (xMort) [reply libs/sdk-backend-tiger/src/backend/workspace/dataFilters/index.ts:15]: I set size to 1000 and added explanation comment.

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/18758/ : SUCCESS in 3s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/20451/ : SUCCESS in 12m 43s

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-fossa-scanning-zuul/2088/ : SUCCESS in 2m 02s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/20454/ : SUCCESS in 12m 35s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-cypress-tiger-isolated-zuul-pipeline/1888/ : SUCCESS in 15m 36s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-cypress-bear-isolated-zuul-pipeline/1870/ : SUCCESS in 6m 30s
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-merge-compliance-check/5384/ : SUCCESS in 6s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-compatibility-pipeline/712/ : SUCCESS in 9m 26s

**Pull request merged**